[FE] feat: 리뷰 목록 페이지에 변경된 API 연동 후, 리액트 쿼리 적용 및 무한 스크롤 구현

soosoo22 opened the pull request:
- resolves #140 

---

### 🚀 어떤 기능을 구현했나요 ?
리뷰 목록페이지에 변경된 API 연동 후, 리액트 쿼리를 적용하고 무한 스크롤을 구현했어요.


### 🔥 어떻게 해결했나요 ?

### 변경된 API 적용

`size, lastReviewId`가 없어지고 `revieweeName, projectName`이 추가되었습니다.
```tsx
export interface ReviewPreviewList {
  revieweeName: string;
  projectName: string;
  reviews: ReviewPreview[];
}

export interface ReviewPreview {
  id: number;
  createdAt: string;
  contentPreview: string;
  keywords: Keyword[];
}
```

getReviewListApi를 사용할 때 groupAccessCode를 인자로 받고 헤더에 groupAccessCode를 포함시킵니다.
```tsx
export const getReviewListApi = async (groupAccessCode: string) => {
  const response = await fetch(endPoint.gettingReviewList, {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
      GroupAccessCode: groupAccessCode,
    },
  });

  if (!response.ok) {
    throw new Error(createApiErrorMessage(response.status));
  }

  const data = await response.json();
  return data as ReviewPreviewList;
};
```

### 리액트 쿼리

- `ReviewPreviewListPage`에서 리액트 쿼리의 `useInfiniteQuery` 훅을 사용해서 리뷰 목록 데이터를 비동기적으로 가져오도록 했습니다.
- 리뷰 목록 API 호출 함수인 `getReviewListApi`를 사용해서 데이터를 불러왔습니다.

`src/index.tsx` ⇒ 리액트 쿼리 적용을 위해 `QueryClientProvider` 추가했습니다.
```tsx
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';


const queryClient = new QueryClient();

// 중간 코드 생략...


root.render(
  <React.StrictMode>
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={theme}>
        <Global styles={globalStyles} />
        <RouterProvider router={router} />
      </ThemeProvider>
    </QueryClientProvider>
  </React.StrictMode>,
);
```


### 무한 스크롤
- `useInfiniteQuery` 훅을 사용해서 무한 스크롤 기능을 구현했습니다.
- 페이지의 마지막 리뷰가 화면에 보일 때 자동으로 다음 페이지의 리뷰를 불러오기 위해 `IntersectionObserver`를 설정했습니다.

### 📝 어떤 부분에 집중해서 리뷰해야 할까요?
api가 계속 변경될 것 같아서 일단은 msw를 사용해서 목 데이터 기반으로 무한 스크롤 기능을 테스트해 주시기 바랄게요!

```tsx
// src/index.tsx에 전 코드 주석 처리 후 아래 코드를 추가
async function enableMocking() {
  if (process.env.MSW) {
    const { worker } = await import('./mocks/browser');
    return worker.start();
  }
}

enableMocking().then(() => {
  root.render(
    <React.StrictMode>
      <QueryClientProvider client={queryClient}>
        <ThemeProvider theme={theme}>
          <Global styles={globalStyles} />
          <RouterProvider router={router} />
        </ThemeProvider>
      </QueryClientProvider>
    </React.StrictMode>,
  );
});
```

### 📚 참고 자료, 할 말
- 레벨 2 미션 상품 목록 코드를 참고했습니다.
- 무한스크롤 정말 어렵네요~~^^

soosoo22 committed:
design: 리뷰 미리보기 카드 배경색 변경 및 전체 감싸는 div에 아래쪽 여백 추가

soosoo22 committed:
refactor: 리뷰 목록 인터페이스 추가 및 수정

- ReviewPreviewList 인터페이스 추가
  - size, lastReviewId, reviews 필드 추가
- ReviewPreview 인터페이스 수정
  - keywords 필드를 Keyword[] 타입으로 변경

soosoo22 committed:
chore: 닫힌 자물쇠로 아이콘 변경

soosoo22 committed:
feat: 리뷰 목록 페이지에서 리액트 쿼리 적용

soosoo22 committed:
refactor: 웹 접근성을 위한 alt, aria-label 추가 및 코드 컨벤션에 맞춰 코드 수정

soosoo22 committed:
feat: 리뷰 미리보기 리스트 핸들러에 페이지네이션 로직 추가

- 기존의 단순 반환 핸들러에서 페이지네이션 기능을 포함한 핸들러로 수정

soosoo22 committed:
feat: `SideModal` 컴포넌트에 `closeModal` prop 추가

soosoo22 committed:
feat: 리뷰 목록 페이지에 무한 스크롤 기능 추가

soosoo22 committed:
refactor: useReviewPreviewList 훅 생성하여 데이터 호출 로직 분리

soosoo22 committed:
refactor: 불필요한 ButtonContainer 제거

donghoony committed:
[All] infra: 디스코드 웹훅 사용해 PR 생성/코멘트 시 멘션 (#169)

* feat: PR Request 생성 / Comment 시 디스코드 멘션

* fix: Case-sensitive ID로 해결, ALL webhook 추가

* fix: remove whitespace

* chore: rename workflow

* fix: fix shell script

* fix: step statement

* fix: remove trailing whitespace after equals sign

BadaHertz52 commented on frontend/src/components/ReviewPreviewCard/index.tsx:
```diff
@@ -1,40 +1,25 @@
+const ReviewPreviewCard = ({ id, reviewerGroup, createdAt, contentPreview, keywords, isPublic }: ReviewPreview) => {
   return (
-    <S.Layout>
+    <S.Layout data-id={id}>
```
data-id를 사용한 이유가 있을까요?

data-id가 개발자 도구를 켰을때 보이는 거라 서버에서 받은 데이터가 개발자 도구에 바로 보일 거라 보안상 우려가되네요.

soosoo22 replied:
https://github.com/woowacourse-teams/2024-review-me/pull/164#discussion_r1699445639 

아하.. 보안 이슈도 있군요..
에프이에게 비슷한 질문을 받아서 해당 답변을 읽어보면 될 것 같아요!

donghoony committed:
[BE] feat: 리뷰 그룹 생성 API 구현 (#163)

* feat: 랜덤한 문자열 생성기 구현

Co-authored-by: donghoony <aru0504@naver.com>

* feat: 리뷰 그룹 생성 요청, 응답 형식

Co-authored-by: donghoony <aru0504@naver.com>

* feat: 리뷰 그룹 생성

Co-authored-by: hyeonjilee <skylar12200@gmail.com>
Co-authored-by: nayonsoso <panda0329@naver.com>

* feat: 리뷰 작성 컨트롤러

Co-authored-by: hyeonjilee <skylar12200@gmail.com>
Co-authored-by: nayonsoso <panda0329@naver.com>

---------

Co-authored-by: nayonsoso <panda0329@naver.com>
Co-authored-by: hyeonjilee <skylar12200@gmail.com>

BadaHertz52 commented on frontend/src/components/ReviewPreviewCard/index.tsx:
```diff
@@ -1,40 +1,25 @@
 import GithubLogo from '@/assets/githubLogo.svg';
 import Lock from '@/assets/lock.svg';
```
svg를 Icon을 뒤에 붙여주세요.

soosoo22 replied:
또또또 까먹었네요....😭

BadaHertz52 commented on frontend/src/components/ReviewPreviewCard/index.tsx:
```diff
@@ -1,40 +1,25 @@
+        </S.HeaderContent>
         <S.Visibility>
-          <img src={isPublic ? UnLock : Lock} />
+          <img src={isPublic ? UnLock : Lock} alt="자물쇠 아이콘" />
           <span>{isPublic ? '공개' : '비공개'}</span>
```
alt가 리뷰 공개/비공개에 따라 달라지면 좋겠어요.

리뷰 공개/비공개에 따른 src,alt, span의 text를 객체로 만들어서 관리하면 좋을것 같아요.

BadaHertz52 commented on frontend/src/components/ReviewPreviewCard/styles.ts:
```diff
@@ -11,10 +11,10 @@ export const Layout = styled.div`
   &:hover {
     cursor: pointer;
-    border: 1px solid ${({ theme }) => theme.colors.primaryHover};
+    border: 1px solid ${({ theme }) => theme.colors.lightPurple};
```
'px' 또 찾았다!!

soosoo22 replied:
전체검색에서 `px` 찾아서 다 고쳤습니다~~~

Kimprodp committed:
[BE] refactor: 리뷰 작성을 위한 정보 요청 기능 리팩터링 (#162)

* refactor: 불필요한 update 방지

* feat: controller에 리뷰 생성 요청에 대한 응답 기능 구현

* feat: controller에 리뷰 작성을 위해 필요한 정보를 응답 기능 구현

* feat: service에 리뷰 작성을 위해 필요한 정보 조회
 기능 구현

* chore: 사용하지 않는 dto 삭제

* test: 리뷰 리뷰 작성을 위해 필요한 정보 조회 기능 테스트 작성

* refactor: swagger 설정 일부 수정 및 에러 응답코드 노출되도록 변경

* refactor: service에서 원시타입 long을 반환하도록 변경

* docs: api 문서 dto 항목설명 변경

* test: 사용하지 않는 변수 선언하지 않도록 변경

* docs: api 문서 검증 내용에 대한 어체 변경

* refactor: 트랜잭션 조회 명시 추가

* refactor: dto 이름 변경

BadaHertz52 commented on frontend/src/hooks/useReviewPreviewList.ts:
```diff
@@ -0,0 +1,31 @@
+export const useReviewPreviewList = () => {
+  const { data, fetchNextPage, hasNextPage, isLoading, error } = useInfiniteQuery({
+    queryKey: [QUERY_KEYS.reviews],
+    queryFn: ({ pageParam = 0 }) =>
```
pageParam이 무엇일까요? 
pageParam이 lastReviewId로 사용되는데, 제가 알고 있기로는 서버에서 lastReviewId를 넘겨주지 않으면 빈값이여야해요. 그런데 pageParam의 기본값이 0이라서 서버에서 lastReviewId가 없을 때 빈값이 아닌 0이 lastReviewId 값으로 들어가게 돼요. 그렇다면 실제 서버와 api연동 시 리뷰 목록 불러오기에서 오류가 날것 같아요.

soosoo22 replied:
pageParam이 카드 목록 중 마지막 카드 아이템 id예요. 그래서 pageParam 값을 lastReviewId에 넣었어요.

생각해보니 pageParam이 0이 되면 안되는데 0값을 계속 넣어주고 있었네요ㅠㅠ 고마워요!

BadaHertz52 commented on frontend/src/mocks/handlers/review.ts:
```diff
@@ -6,6 +6,13 @@ import { DETAILED_REVIEW_MOCK_DATA } from '../mockData/detailedReviewMockData';
 import { REVIEW_PREVIEW_LIST } from '../mockData/reviewPreviewList';
 import { REVIEW_WRITING_DATA } from '../mockData/reviewWritingData';
 
+export const PAGE = {
```
리뷰에 대한 여러 api 핸들러가 있는 파일에서 PAGE라는 상수명 대신에 리뷰 목록 리스트에 한정하는 상수명은 어떨까요?

soosoo22 replied:
바다 말대로 작성, 목록, 상세 페이지에 대한 api 핸들러가 있는 파일에서는 리뷰 목록 리스트에 대한 상수명으로 정의하는 게 좋을 것 같네요! 고마워요!

BadaHertz52 commented on frontend/src/mocks/handlers/review.ts:
```diff
@@ -16,10 +23,31 @@ const getDataToWriteReview = () =>
+
+    const lastReviewId = Number(url.searchParams.get('lastReviewId'));
+
+    const isFirstPage = lastReviewId === 0;
```
첫페이지라면 lastReviewId값이 없는거라고 알고 있는데,여기에도 lastReviewId의 기본값을 0으로 사용하고 있네요.

만약 0이 맞더라도 0이 매직넘버라 상수로 분리해주세요.

BadaHertz52 commented on frontend/src/mocks/handlers/review.ts:
```diff
@@ -16,10 +23,31 @@ const getDataToWriteReview = () =>
+    const limit = isFirstPage ? PAGE.defaultPageSize : PAGE.additionalPageSize;
+    const startIndex = isFirstPage
+      ? PAGE.firstPageStartIndex
+      : REVIEW_PREVIEW_LIST.reviews.findIndex((review) => review.id === lastReviewId) + 1;
```
1이라는 매직넘버 발견! 

코드가 많아서 상수를 분리하는 것을 놓칠 수 있죠
이렇게 놓치는 것을 잡아주는데 코드 리뷰의 순기능일지도?
😊!!

BadaHertz52 commented on frontend/src/pages/ErrorPage/components/ErrorSection/index.tsx:
```diff
@@ -37,16 +37,16 @@ const ErrorSection = ({ errorMessage, handleReload, handleGoHome }: ErrorSection
-        {buttons.map((button, index) => (
-          <S.ButtonContainer key={index}>
+        {buttons.map((button) => (
+          <>
```
쑤쑤 ErrorSection의 변경 사항 제꺼에 반영했어요.
ButtonContainer가 삭제되면서 flagment 가 생겼는데 이유는 뭘까요?

chysis commented on frontend/src/components/ReviewPreviewCard/styles.ts:
```diff
@@ -11,10 +11,10 @@ export const Layout = styled.div`
   &:hover {
     cursor: pointer;
-    border: 1px solid ${({ theme }) => theme.colors.primaryHover};
+    border: 1px solid ${({ theme }) => theme.colors.lightPurple};
```
rem 단위로 통일시키면 좋을 것 같아요👀

soosoo22 replied:
아이고~~~~ 또 `px`을 썼네요😭
다시는 이런 실수를 반복하지 않겠습니다!!!!!!

BadaHertz52 commented on frontend/src/pages/ReviewPreviewListPage/index.tsx:
```diff
@@ -15,53 +15,57 @@ const OPTIONS = ['최신순', '오래된순'];
-                />
-              </div>
-            ))}
+          {isLoading && <LoadingPage />}
+          {error && <p>{error.message}</p>}
```
loading,error 에 Suspense, ErrorBoundary를 아직 적용이 안되는 건가요?

error시에도 error.message를 띄워주니 ErrorSection처럼 피그마에 적용한 오류 시안을 ui로 적용하고 있지않네요.

BadaHertz52 commented on frontend/src/pages/ReviewPreviewListPage/index.tsx:
```diff
@@ -15,53 +15,57 @@ const OPTIONS = ['최신순', '오래된순'];
+
+  const lastReviewElementRef = useCallback(
+    (node: HTMLElement | null) => {
+      if (isLoading) return console.log('isLoading', isLoading);
```
질문1: console.log는 무한 스크롤을  확인하기 용도인가요? 그렇다면 push에서는 console.log를 지워주어야 하지 않나요?

질문2: 무한 스크롤로 데이터를 불러오는 동안, lastReviewElementRef에서 로딩 중임을 보여주는 UI가 필요해요.

soosoo22 replied:
질문1 답변 : 무한스크롤 확인하려고 넣었던 코드인데 지우질 않았네요😅 

질문2 답변 : 그 부분을 제가 놓치고 있었네요. `LoadingBar` 컴포넌트를 넣을게요!

BadaHertz52 commented on frontend/src/types/review.ts:
```diff
@@ -60,6 +60,12 @@ export interface WritingReviewInfoData {
 
+export interface ReviewPreviewList {
+  size: number;
+  lastReviewId: number;
```
lastReviewId은 타입도 number로 되어있어서 이거를 수정해야겠네요.

BadaHertz52 reviewed (commented):
일단 무스를 끝낸 것에 박수를 보내요.👏👏👏👏👏👏
쑤쑤 리뷰 남겼어요.

BadaHertz52 reviewed (changes requested):
쑤쑤 우선 무스 기능 구현을 성공한 것에 박수를 보내요.👏👏👏👏👏👏👏👏  역시 쑤쑤가 해낼 줄 알았지

쑤쑤의 성장을 바라며 코멘트를 남겨뒀어요. 확인 부탁드려요.

chysis commented on frontend/src/pages/ErrorPage/components/ErrorSection/index.tsx:
```diff
@@ -37,16 +37,16 @@ const ErrorSection = ({ errorMessage, handleReload, handleGoHome }: ErrorSection
-        {buttons.map((button, index) => (
-          <S.ButtonContainer key={index}>
+        {buttons.map((button) => (
+          <>
             <Button
               buttonType={button.buttonType}
               text={button.text}
               image={button.image}
               imageDescription={button.imageDescription}
               onClick={button.onClick}
             />
-          </S.ButtonContainer>
+          </>
```
<S.ButtonContainer> 내부에서 `map`을 사용하고, `Button` 컴포넌트에 key를 추가하는 것은 어떨까요?

```tsx
<S.Container>
  <S.ButtonContainer>
    {buttons.map((button) => (
      <Button
        // ...
      />
    )}
  </S.ButtonContainer>
</S.Container>
```

soosoo22 replied:
아하 `key`를 추가하는 게 좋겠네요!
`<S.ButtonContainer>`는 불필요해서 제거했고 `<S.Container>`안에서 `map`을 돌리고 `Button`에 `key`를 추가했어요!

```tsx
 <S.Container>
        {buttons.map((button, index) => (
          <>
            <Button
              key={index}
              buttonType={button.buttonType}
              text={button.text}
              image={button.image}
              imageDescription={button.imageDescription}
              onClick={button.onClick}
            />
          </>
        ))}
      </S.Container>
```

chysis commented on frontend/src/components/ReviewPreviewCard/index.tsx:
```diff
@@ -1,40 +1,25 @@
+const ReviewPreviewCard = ({ id, reviewerGroup, createdAt, contentPreview, keywords, isPublic }: ReviewPreview) => {
   return (
-    <S.Layout>
+    <S.Layout data-id={id}>
```
data attribute를 사용한 이유가 궁금해요!

soosoo22 replied:
```tsx
export interface ReviewPreview {
  id: number;
  isPublic: boolean;
  reviewerGroup: {
    id: number;
    name: string;
    thumbnailUrl: string;
  };
  createdAt: string;
  contentPreview: string;
  keywords: Keyword[];
}

```

`ReviewPreviewCard` 컴포넌트에는 원래 `reviewerGroup, createdAt, contentPreview, keywords, isPublic`만 필요했지만, 새로운 인터페이스를 정의하지 않고 기존의 `ReviewPreview` 인터페이스를 재사용하기 위해 `id`를 추가했어요. `id`를 사용할 자리가 없어 `data-id` 속성을 활용했어요.

BadaHertz52 replied:
```
data-id={id}
```
서버에서 받아온 리뷰의 id를 data-id에 써서, html에서 리뷰 id가 노출될 것 같아요.
data-id가 어디에서 사용되고 있는건가요?

soosoo22 replied:
`ReviewPreviewCard` 컴포넌트에서 사용되고 있어요. id 제거할게요!

chysis reviewed (changes requested):
미션 할 때 무한 스크롤 때문에 고생했던 기억이 눈에 선하네요🥲 정말 고생 많았어요
`useInfiniteQuery`를 사용한 부분은 함께 이야기 나눈 부분이라 별도로 피드백을 남기진 않았어요. 코멘트 확인 부탁해요!

ImxYJL commented on frontend/src/components/ReviewPreviewCard/styles.ts:
```diff
@@ -27,11 +27,10 @@ export const Header = styled.div`
+  border-radius: 0.8rem 0.8rem 0 0;
 `;
 
-export const HeaderContainer = styled.div`
```
Container에서 Content로 이름을 바꾼 이유가 궁금합니다!

soosoo22 replied:
`Layout` 안에 `Header, Main`을 기준으로 `Header`안에 `HeaderContainer`가 있는 게 어색한 것 같아서 `HeaderContent`로 바꿨어요!

```html
<S.Layout>
  <S.Header>
    <S.HeaderContent />
  </S.Header>
  <S.Main />
</S.Layout>
```

ImxYJL commented on frontend/src/mocks/handlers/review.ts:
```diff
@@ -6,6 +6,13 @@ import { DETAILED_REVIEW_MOCK_DATA } from '../mockData/detailedReviewMockData';
+};
+
 const getDetailedReview = () =>
   http.get(endPoint.gettingDetailedReview(123456, 123456), async ({ request }) => {
```
이렇게 123456을 사용하는 부분도 가능하다면 상수화나 간단한 주석이 있으면 나중에 알아보기 더 편할 것 같아요!

soosoo22 replied:
아하! 고마워용!!! 상수처리 너무 중요하죠!

ImxYJL commented on frontend/src/mocks/handlers/review.ts:
```diff
@@ -16,10 +23,31 @@ const getDataToWriteReview = () =>
-const getReviewPreviewList = () =>
-  http.get(endPoint.gettingReviewList(1, 3, 1), async ({ request }) => {
-    return HttpResponse.json(REVIEW_PREVIEW_LIST);
+const getReviewPreviewList = () => {
```
이렇게 디테일하게 get 핸들러 함수를 작성한 이유는 무한 스크롤 로직(API가 보내주는 데이터를 프론트에서 의도한 대로 처리하고 있는지)이 정상적으로 작동하는지 확인하기 위함이었나요?

soosoo22 replied:
네 맞아요. 서버에는 리뷰 목록 개수가 많지 않아서 무한스크롤을 제대로 확인하기 힘들어서 get 핸들러 함수에 디폴트 페이지, 추가적으로 불러오는 페이지 개수 등을 구체적으로 명시해서 제대로 기능이 작동하는지 확인했어요

ImxYJL commented on frontend/src/pages/ReviewPreviewListPage/index.tsx:
```diff
@@ -15,53 +15,57 @@ const OPTIONS = ['최신순', '오래된순'];
+      if (isLoading) return console.log('isLoading', isLoading);
+      if (observer.current) observer.current.disconnect();
+
+      observer.current = new IntersectionObserver((entries) => {
```
지금은 바빠서 무리겠지만 ㅎㅎ;
`IntersectionObserver`를 이용한 무한 스크롤 로직은 별도의 훅으로 분리해보면 좋을 것 같아요.
제가 예전에 무한스크롤 훅에 대한 글을 썼었는데 참고해보세용~~ [이거](https://velog.io/@allensain14/IntersectionObserver-with-React-Typescript)

soosoo22 replied:
추후 리팩할 부분 중 하나였는데 짚어주셔서 너무 감사해요!!!
참고자료까지 너무 고마워요 올리❤️

ImxYJL reviewed (approved):
무한 스크롤 끝까지 붙잡고 있던 쑤쑤에게 박수~
크리티컬한 건 이미 바다가 잡아줘서 저는 비교적 간단한 코멘트만 남기고 어프루브 할게요!

soosoo22 committed:
design: px을 rem으로 변경

soosoo22 committed:
refactor: Button 컴포넌트에 key 추가

BadaHertz52 committed:
[FE] 리뷰 상세 페이지에 react-query 추가 및 리팩토링 (#161)

* fix: src/index.tsx에서 enableMocking 제거

* feat: 리뷰 그룹 생성 시 코멘트 컴포넌트명 변경 및 기본값 설정

- 컴포넌트명 변경: RevewComment -> ReveweeComments
-  reviewGroup의 description이 빈문자열이면 기본값을 보여주는 것으로 변경

* refactor: 서버 DB에 있는 리뷰 데이터를 사용하기 위한 상수화

- 현재 DB에 있는 리뷰 데이터를 목 서버에서도 사용하고, 사이드바 페이지 이동 시에도 활용할 수 있도록 관련 value들을 상수화 함

* feat : QueryClient, QueryClientProvider 적용

- src/index.tsx에 QueryClient, QueryClientProvider 적용

* fix: dependencies에 있는 테스트 패키지들을 devDependencies로 옮김

* feat: 리뷰 상세페이지에 react-query 적용

* feat : msw에 리뷰 상세페이지 404 오류 추가

* ci: react-error-boundary 설치

* feat: Outlet에 QueryErrorResetBoundary,ErrorBoundary, Suspense 적용

* feat: 리뷰 상세페이지에 useSuspenseQuery 적용

* refactor: 리뷰 상세페이지 resource, queryString key 상수화

* refactor:  리뷰 상세페이지 react-query key 상수화

* refactor: 리뷰 상세 페이지 컴포넌트 속에서만 사용하는 상수들 상수화

* refactor: DetailedReviewPage/components에 index.tsx를 추가해 import 경로 간결하게 수정

* feat: error 전파를 위한 QueryClient 옵션 추가

-  react-query의 query, mutation에서 error가 전파되도록 QueryClient 옵션 설정

* fix: ErrorPage의 SideModal에 closeModal props로 줌

* refactor: ErrorSection 위치 변경(src/pages/ErrorPage -> src/components/error)

* feat: ErrorFallback 컴포넌트 생성

* feat: ErrorSuspenseContainer 생성 및 App.tsx에 적용

* chore: constants/index.ts export 경로 변경

- 중복되는 apiErrorMessage 삭제
- queryKeys 추가

* chore: 3차-1주차 핵심 기능 시현 때 필요 없는 코드 주석 처리

* docs: ErrorPage의 ERROR_MESSAGE 수정

* design: formWidth 변경 및 fontSize에 1.4rem 추가

* feat: 리뷰 상세 페이지에 리뷰이 이름 추가

- 리뷰 상세 페이지 목데이터, 데이터 타입에 리뷰이 이름 추가
- 리뷰 상세 페이지 컴포넌트에 리뷰이 이름 추가 및 관련 스타일 추가

* refactor: 불필요한 export 삭제

* chore: type명 수정 (RevieweeCommentProps =>RevieweeCommentsProps)

* refactor: ErrorSection으l Button 수정

* refactor: 리뷰 상세 페이지 데이터 타입 변경에 따른 수정

* refactor: ErrorSuspenseContainer 적용 위치 변경

- App가 아닌 router의 element에서 적용하는 것으로 변경

* refactor: 리뷰 상세 페이지 데이터 타입 강제 방법 변경

* chore: 불필요한 주석 삭제

* refactor: ErrorSection의 buttons 네이밍 변경 및 요소에 key 추가

- buttons -> buttonList

* chore: 스타일 주석에 NOTE 추가

donghoony committed:
[BE] feat: 리뷰 목록 조회  (#179)

* feat: 리뷰 미리보기

Co-authored-by: nayonsoso <panda0329@naver.com>

* feat: 내가 받은 리뷰 목록 응답 생성

Co-authored-by: donghoony <aru0504@naver.com>

* feat: 리뷰 목록 조회

Co-authored-by: nayonsoso <panda0329@naver.com>

* refactor: Cascade 적용으로 불필요한 save 제거

Co-authored-by: nayonsoso <panda0329@naver.com>

* refactor: 리뷰 미리보기 생성 객체 도출

Co-authored-by: nayonsoso <panda0329@naver.com>

---------

Co-authored-by: nayonsoso <panda0329@naver.com>

BadaHertz52 commented on frontend/src/components/ReviewPreviewCard/index.tsx:
```diff
@@ -1,40 +1,25 @@
+        </S.HeaderContent>
         <S.Visibility>
-          <img src={isPublic ? UnLock : Lock} />
+          <img src={isPublic ? UnLock : Lock} alt="자물쇠 아이콘" />
```
alt가 비공개/공개 여부에 따라 바뀌면 alt을 사용하는 의미가 더 명확해질 것 같아요.

chysis committed:
fix: 브라우저 타이틀 오타 수정 (#167)

Kimprodp committed:
[BE] feat: 리뷰 상세 조회 기능 구현 (#182)

* feat: 컨트롤러 리뷰
상세 조회 api 응답 기능 구현

* feat: 리뷰 상세 조회 api 응답 객체 구현

* refactor: 리뷰 상세 조회 api 응답 객체명 변경

* feat: 서비스 리뷰 상세 조회 기능 구현

* test: 리뷰 상세 조회 기능 테스트 추가

* style: 오타 및 개행 수정

* refactor: 날짜 응답 시, 일자까지만 응답하도록 변경

* refactor: 리뷰 조회 권한 검증 로직을 repository 사용하도록 변경

* refactor: 검증메세지 변경

* refactor: 리뷰 서비스에서 리뷰 그룹을 찾을 수 없는 경우에 대한 예외 클래스 변경

* refactor: 리뷰 상세 조회 메서드명 변경

* refactor: 헤더명 상수화

BadaHertz52 committed:
[FE] recoil을 통한 groupAccessCode 전역 상태 관리 (#180)

* ci: recoil 설치

* feat: src/index.ts에 RecoilRoot 적용

* feat: groupAccessCode 에 대한 atom 상태 추가

* feat: useGroupAccessCode 훅 추가

- useGroupAccessCode : groupAccessCode의 atom 상태를 관리하는 훅

nayonsoso committed:
[BE] 내가 받은 리뷰 목록을 최신순으로 조회한다. (#189)

* fix: 날짜 순 정렬

* refactor: 사용하지 않는 변수 제거

---------

Co-authored-by: donghoony <aru0504@naver.com>

Kimprodp committed:
[BE] 초기 데이터 설정 (#187)

* feat: 초기 데이터 설정

* build: CD 시 local profile 설정

---------

Co-authored-by: donghoony <aru0504@naver.com>

chysis committed:
[FE] feat: Alert Modal과 Error Alert Modal 컴포넌트 구현 및 Button 컴포넌트 확장 (#165)

* refactor: 공통 버튼 컴포넌트가 children과 style을 받도록 수정

* chore: error alert modal에서 사용할 primary 색상의 경고 삼각형 추가

* feat: ErrorAlertModal 컴포넌트 구현

* feat: AlertModal 컴포넌트 구현

* chore: 변경된 Button 컴포넌트의 구조에 맞게 코드 수정

* fix: AlertModal이 Esc 또는 background의 클릭으로 닫히지 않도록 수정

* chore: AlertModal과 ErrorAlertModal export

* refactor: Button 컴포넌트가 type을 별도로 받도록 수정 및 기존의 buttonType을 styleType으로 변경

* refactor: background 클릭 또는 Esc 키를 통해 모달을 닫는 것을 분리

* refactor: AlertModal의 모달 닫는 정책을 props로 선택 가능하도록 수정

* refactor: ErrorAlertModal을 AlertModal을 사용해 구현

* refactor: 닫기 버튼 이외의 방법으로 모달을 닫을 수 있는지 여부와 handler를 props로 전달받도록 변경

* refactor: 비어있는 스타일 컴포넌트 제거

soosoo22 committed:
feat: 리뷰이, 프로젝트명을 보여주는 DescriptionSection 컴포넌트 구현

soosoo22 committed:
refactor: 리뷰 목록을 불러오는 api 변경 이슈로 인해 엔드포인트, fetch 함수 수정

soosoo22 committed:
chore: fragment 제거

soosoo22 committed:
refactor: api 변경에 따라 ReviewPreviewList, ReviewPreview 인터페이스 수정

ImxYJL committed:
[FE] 랜딩 페이지 퍼블리싱 및 디바운싱 함수 추가 (#181)

* feat: Input 공통 컴포넌트 제작

* feat: LandingPage 제작

* fix: Input의 type을 text로 수정

* feat: debounce 함수 작성

* docs: debounce 함수에 TODO 주석 및 TSdoc 추가 외 간단한 if문 리팩토링

* chore: 잘못된 파일 구조 및 디렉토리명 수정

* chore: px을 rem으로 수정

* refactor: 동적 스타일링에 css 대신 styled 스타일 적용

* refactor: Input 스타일을 더욱 유연하게 받을 수 있도록 수정, 스타일 리터럴 제거

* refactor: Input에 적용된 커스텀 스타일링을 $style로 수정

* fix: undefined 리턴을 null 리턴으로 수정

* refactor: flex-direction에 별도의 커스텀 타입을 사용하는 대신 React 내장 타입을사용하도록 수정

* refactor: Input 컴포넌트에서 rest props 제거

* chore: 불필요한 import 제거

soosoo22 committed:
refactor: 변경된 api에 따라 props 수정

soosoo22 committed:
refactor: reviewerGroup, isPublic 제거

soosoo22 committed:
refactor: endPoint.gettingReviewList 호출 인자 변경

soosoo22 committed:
refactor: DescriptionSection에서 ReviewInfoSection으로 컴포넌트명 변경

soosoo22 committed:
refactor: useSuspenseQuery 적용 후, 무한스크롤 관련 코드 주석 처리

BadaHertz52 committed:
Merge branch 'fe/feat/140-review-list-react-query' of https://github.com/woowacourse-teams/2024-review-me into fe/feat/140-review-list-react-query

BadaHertz52 reviewed (approved):
쑤쑤 고생 많았어요.

chysis reviewed (approved):
저엉말 고생많았어요😵‍💫👍

ImxYJL reviewed (approved):
우하하

BadaHertz52 reviewed (approved):
😝

ImxYJL reviewed (approved):
돼라!

donghoony commented:
수수 최고..
